src/ng1/directives/treegrid/treegrid.html

```diff
@@ -1,45 +1,89 @@
-              <i ng-if="row.expander.type === 'class'" ng-class="vm.iconClass(row.expander.expanding)"></i>
-              <img ng-if="row.expander.type === 'url'" ng-src="{{row.expander.expanding}}" alt="expander" />
-            </span>
-          </span>
-          <span ng-if="$first && vm.allOptions.select.check" class="treegrid-checkbox">
-            <checkbox class="treegrid-multiple-select-checkbox" ng-model="row.selected" ng-click="vm.checkboxClick($event);"></checkbox>
-          </span>
-          <span ng-if="$first" class="treegrid-icon">
-            <i ng-if="row.icon.type === 'class'" class="hpe-icon" ng-class="row.icon.get(row.dataItem, row.expanded)"></i>
-            <img ng-if="row.icon.type === 'url'" ng-src="{{row.icon.get(row.dataItem, row.expanded)}}" alt="icon" />
-          </span>
-          <treegrid-cell row="row" column="col"></treegrid-cell>
-        </td>
-      </tr>
-    </tbody>
-  </table>
-  <div ng-if="vm.loading" class="treegrid-loading">
-    Loading...
-  </div>
+
+    <table class="table table-hover treegrid" keyboard-navigable-table tabindex="0" ng-focus="vm.focusFirstRow()">
+
+        <thead>
+            <tr class="table-header-dark">
+
+                <th ng-repeat="column in vm.columns"
+                    ng-class="column.headerClass"
+                    ng-style="{ 'width': column.width }">
+
+                    <span ng-if="$first && vm.allOptions.select.check"
+                          class="treegrid-header-checkbox">
+
+                        <checkbox class="treegrid-multiple-select-checkbox"
+                                  ng-model="selected"
+                                  ng-click="vm.toggleAllRows($event); $event.preventDefault();">
+                        </checkbox>
+                    </span>
+
+                    <span ng-class="[$first ? 'treegrid-expand-header' : '', vm.allOptions.select.check ? 'has-checkbox' : '']"
+                          ng-bind="column.name">
+                    </span>
+                </th>
+            </tr>
+
+        </thead>
+
+        <tbody ng-if="!vm.loading" role="tree">
+
+            <tr ng-repeat="row in vm.getGridRows()"
+                class="clickable hover-actions"
+                ng-class="[row.rowClass, row.selected ? vm.allOptions.select.rowClass : '', row.level > 0 ? 'sub-row' : '']"
+                treegrid-multiple-select-item="row"
+                treegrid-multiple-select-options="vm.allOptions.select"
+                treegrid-row-key-handler
+                treegrid-expand="vm.expand(row)"
+                treegrid-contract="vm.contract(row)"
+                ng-focus="vm.rowFocus(row, $event)"
+                aria-expanded="{{ row.expanding }}"
+                aria-checked="{{ row.selected }}"
+                role="{{ row.canExpand ? 'group' : 'treeitem' }}"
+                tabindex="-1">
+
+                <td ng-repeat="col in vm.columns" ng-class="col.cellClass">
```

Comment:
It might help to add aria-labels with the column name on each td, since otherwise Chromevox just reads the whole row out in one go. (Unless `role="treegrid"` improves this.)